src/native/external/zlib/dotnet_allocator_unix.c

```diff
@@ -0,0 +1,162 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdint.h>
+#include "zutil.h"
+
+/* A custom allocator for zlib that provides some defense-in-depth over standard malloc / free.
+ * (non-Windows version)
```

Comment:
In the current repo setup, this is only going to be used for Wasm. It won't kick in for Linux, macOS, etc. Does it match your expectations?

Reply:
Yes, this matches my expectations. If there is some future distro we want to support where that distro does not include a shared zlib module, then this would also kick in.

This ties back to a previous conversation, where I understand we do not want to carry our own customizations to zlib on platforms where zlib is natively provided and where admins are expected to keep it up-to-date.

Reply:
Yes, this matches my expectations. If there is some future distro we want to support where that distro does not include a shared zlib module, then this would also kick in.

This ties back to a previous conversation, where I understand we do not want to carry our own customizations to zlib on platforms where zlib is natively provided and where admins are expected to keep it up-to-date.